apps/chain-tool/main.cpp

```diff
@@ -0,0 +1,890 @@
+           "have inconsistent(non-continuous) numbering -> there are missing files for one or more "
+           "indexes.";
+      return {std::move(tx_stores), -2, s.str()};
+    }
+
+    if (!meta::IsLog2(tx_stores.size()))
+    {
+      std::ostringstream s;
+      s << "Inferred number of lanes " << tx_stores.size()
+        << " (number of file indexes) MUST be power of 2.";
+      return {std::move(tx_stores), -3, s.str()};
+    }
+
+    // std::cout << "INFO: Inferred number of lanes: " << tx_stores.size() << std::endl;
+  }
+
+  return {std::move(tx_stores), 0, ""};
+}
+
+void ProcessTransactions(BlockChainForwardTree const &bch, BlockChain const &heaviest_chain,
+                         TxStores &tx_stores, TxStoresPtr trimmed_tx_stores,
+                         bool const print_missing_txs)
+{
+
+  constexpr std::size_t num_of_progress_steps{10ull};
+  auto const            num_of_lanes{tx_stores.size()};
+  auto const            log2_num_of_lanes{meta::Log2(num_of_lanes)};
+  std::size_t const     progress_step{(heaviest_chain.num_of_all_txs + num_of_progress_steps - 1) /
+                                  num_of_progress_steps};
+
+  std::size_t tx_count_in_blockchain{0};
+  std::size_t last_reported_progress_tx_count{0};
+  std::size_t tx_count_missing_accumulated{0};
+  std::size_t tx_count_processed{0};
+  std::size_t tx_count_stored_in_trimmed_db{0};
+  std::size_t count_of_all_tx_in_db{0};
+
+  std::vector<uint64_t> tx_count_missing(tx_stores.size(), 0ull);
+
+  for (auto const &tx_lane_store : tx_stores)
+  {
+    count_of_all_tx_in_db += tx_lane_store.second.size();
+    std::cout << "Lane" << tx_lane_store.first
+              << ": Tx Count reported by Tx db store: " << tx_lane_store.second.size() << std::endl;
+  }
+
+  std::cout << "INFO: Checking Transactions from all blocks ... " << std::endl;
+  bch.IterateChainBackward(
+      heaviest_chain,
+      [&tx_stores, &tx_count_in_blockchain, &last_reported_progress_tx_count, &tx_count_missing,
+       &tx_count_missing_accumulated, &tx_count_processed, &tx_count_stored_in_trimmed_db,
+       &trimmed_tx_stores, progress_step, print_missing_txs,
+       log2_num_of_lanes](BlockNode const &node, BlockHash const & /*block_hash*/) {
+        uint64_t slice_idx{0};
+        for (auto const &slice : node.db_record.block.body.slices)
+        {
+          if (!node.is_block_set)
+          {
+            return false;
+          }
+
+          tx_count_in_blockchain += slice.size();
+
+          uint64_t tx_idx_in_slice{0};
+          for (auto const &tx_layout : slice)
+          {
+            auto const  lane{ResourceID{tx_layout.digest()}.lane(log2_num_of_lanes)};
+            Transaction tx;
+
+            bool res{false};
+            try
+            {
+              res = tx_stores[lane].Get(storage::ResourceID{tx_layout.digest()}, tx);
+            }
+            catch (...)
+            {
+              if (print_missing_txs)
```

Comment:
This printout should either be removed (since printout is handled bellow the catch block), or the mesage should be changed - that Tx fetch failed with exception (adding exception`s `what()` message).